api/handlers_metrics.go

```diff
@@ -324,3 +324,159 @@ func parseQuery(r *http.Request, request cloudwatch.MetricsRequest) error {
 
 	return nil
 }
+
+// GetRDSMetricsURLHandler gets RDS metrics from cloudwatch and returns a link to the image
+func (s *server) GetRDSMetricsURLHandler(w http.ResponseWriter, r *http.Request) {
+	w = LogWriter{w}
+	vars := mux.Vars(r)
+	account := vars["account"]
+	instanceId := vars["id"]
+
+	cwService, ok := s.cloudwatchServices[account]
+	if !ok {
+		msg := fmt.Sprintf("cloudwatch service not found for account: %s", account)
+		handleError(w, apierror.New(apierror.ErrNotFound, msg, nil))
+		return
+	}
+	log.Debugf("found cloudwatch service %+v", cwService)
+
+	resultCache, ok := s.resultCache[account]
+	if !ok {
+		msg := fmt.Sprintf("result cache not found for account: %s", account)
+		handleError(w, apierror.New(apierror.ErrNotFound, msg, nil))
+		return
+	}
+	log.Debugf("found cost explorer result cache %+v", *resultCache)
+
+	queries := r.URL.Query()
+	metrics := queries["metric"]
+	if len(metrics) == 0 {
+		handleError(w, apierror.New(apierror.ErrBadRequest, "at least one metric is required", nil))
+		return
+	}
+
+	req := cloudwatch.MetricsRequest{}
+	if err := parseQuery(r, req); err != nil {
+		handleError(w, apierror.New(apierror.ErrBadRequest, "failed to parse query", err))
+		return
+	}
+
+	key := fmt.Sprintf("%s/%s/%s/%v/%v/%v", Org, instanceId, strings.Join(metrics, "-"), req["start"], req["end"], req["period"])
```

Comment:
you want to convert this to use the `req.String()` stuff

Reply:
applied